tangelo/tangelo/__init__.py

```diff
@@ -145,8 +148,13 @@ def good(path):
     # to Nones will have been logged).
     newpaths = filter(lambda x: x is not None, map(good, runtimepaths))
 
+    # Use the import lock to have some thread safety
+    imp.acquire_lock()
+    # Exclude paths we've already added to the system
+    newpaths = [path for path in newpaths if path not in sys.path]
```

Comment:
No worries - let's leave this as is.  Sometimes I like comprehension syntax better and sometimes I don't.  Go figure :)